sdk/go/common/workspace/loaders.go

```diff
@@ -305,3 +315,51 @@ func LoadPolicyPack(path string) (*PolicyPackProject, error) {
 
 	return &policyPackProject, nil
 }
+
+// checkForEmptyConfig emits a warning for any config values that are `null` in
+// the YAML/JSON. These are currently loaded into a `config.Value` that's
+// interpreted as an empty string when we unmarshal into `ProjectStack.Config`,
+// but we want to change this in the future to maintain nullness.
+// Note that for object values, null is maintained inside objects already.
+//
+// ```yaml
+// # somekey1,2,3 are all `null` in YAML, but `""` in pulumi.
+// somekey1:
+// somekey2: null
+// somekey3: ~
+//
+// # someobj.somekey4 is `null` in both YAML and pulumi
+// someobj:
+//
+//	somekey4:
+//	somekey5: null
+//	somekey6: ~
+//
+// ```
+func checkForEmptyConfig(config map[string]interface{}) {
+	keys := []string{}
+	for k, v := range config {
+		if v == nil {
+			keys = append(keys, k)
+		}
+	}
+
+	if len(keys) == 1 {
+		cmdutil.Diag().Warningf(&diag.Diag{
```

Comment:
I am not sure if it's ok to call `cmdutil.Diag()` from `LoadProjectStackBytes`.

Reply:
Thread a sink